parts/k8s/cloud-init/artifacts/cse_config.sh

```diff
@@ -165,18 +165,20 @@ configureK8s() {
     chmod 0644 "${APISERVER_PUBLIC_KEY_PATH}"
     chown root:root "${APISERVER_PUBLIC_KEY_PATH}"
 
-    AZURE_JSON_PATH="/etc/kubernetes/azure.json"
-    touch "${AZURE_JSON_PATH}"
-    chmod 0600 "${AZURE_JSON_PATH}"
-    chown root:root "${AZURE_JSON_PATH}"
-
-    set +x
     echo "${KUBELET_PRIVATE_KEY}" | base64 --decode > "${KUBELET_PRIVATE_KEY_PATH}"
     echo "${APISERVER_PUBLIC_KEY}" | base64 --decode > "${APISERVER_PUBLIC_KEY_PATH}"
     {{/* Perform the required JSON escaping */}}
     SERVICE_PRINCIPAL_CLIENT_SECRET=${SERVICE_PRINCIPAL_CLIENT_SECRET//\\/\\\\}
     SERVICE_PRINCIPAL_CLIENT_SECRET=${SERVICE_PRINCIPAL_CLIENT_SECRET//\"/\\\"}
-    cat << EOF > "${AZURE_JSON_PATH}"
+
+    if [[ -n "${MASTER_NODE}" ]]; then
+        AZURE_JSON_PATH="/etc/kubernetes/azure.json"
+        touch "${AZURE_JSON_PATH}"
+        chmod 0600 "${AZURE_JSON_PATH}"
+        chown root:root "${AZURE_JSON_PATH}"
+
+        set +x
```

Comment:
this needs to stay befor the private key and service principal manipulations above to avoid printing secrets in CSE logs

Reply:
@CecileRobertMichon  by "this" you mean `set +x` ?  thanks for your comment.

Reply:
yes, `set +x`

It disables printing commands to the logs (in this case the commands themselves contain sensitive information)